pkg/chunked/cache_linux.go

```diff
@@ -488,26 +626,41 @@ func readCacheFileFromMemory(bigDataBuffer []byte) (*cacheFile, error) {
 	if err := binary.Read(bigData, binary.LittleEndian, &digestLen); err != nil {
 		return nil, err
 	}
+
+	bloomFilter, err := readBloomFilter(bigData)
+	if err != nil {
+		return nil, err
+	}
+
 	if err := binary.Read(bigData, binary.LittleEndian, &tagsLen); err != nil {
 		return nil, err
 	}
 	if err := binary.Read(bigData, binary.LittleEndian, &vdataLen); err != nil {
 		return nil, err
 	}
 
+	if err := binary.Read(bigData, binary.LittleEndian, &fnamesLen); err != nil {
+		return nil, err
+	}
 	tags := make([]byte, tagsLen)
 	if _, err := bigData.Read(tags); err != nil {
 		return nil, err
 	}
 
 	// retrieve the unread part of the buffer.
-	vdata := bigDataBuffer[len(bigDataBuffer)-bigData.Len():]
+	remaining := bigDataBuffer[len(bigDataBuffer)-bigData.Len():]
+
+	vdata := remaining[:vdataLen]
+	fnames := remaining[vdataLen:]
 
 	return &cacheFile{
-		tagLen:    int(tagLen),
-		digestLen: int(digestLen),
-		tags:      tags,
-		vdata:     vdata,
+		bloomFilter: bloomFilter,
+		digestLen:   int(digestLen),
+		fnames:      fnames,
+		fnamesLen:   int(fnamesLen),
```

Comment:
(Nit: I’d aesthetically prefer if the type declaration, and the two constructors, were all listing the fields in the same order — and if that order were somehow consistent. Here we have (data, len) for file names, but (len, data) for tags, for example.)